lab_1_keywords_tfidf/main.py

```diff
@@ -48,7 +66,14 @@ def calculate_frequencies(tokens: list[str]) -> Optional[dict[str, int]]:
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not (isinstance(tokens, list) and tokens):
+        return None
+    for token in tokens:
+        if not isinstance(token, str):
+            return None
+        frequency_dict = {i: tokens.count(i) for i in tokens}
```

Comment:
with this level of identation, you are composing a dict as many times as there are tokens

Reply:
done